### PR TITLE
Add local mode execution

### PR DESCRIPTION
**Pull request by ferranbt**
## 📝 Summary

This PR adds a `--local` flag to the `suave-geth forge` command. If the flag is set, the command will resolve the precompile logic locally and it will not try to call a running `suave-geth` instance.

This works for all the precompiles except for the confidential store ones since they are stateless. If any of those precompiles is executed in this local mode, it will return an error.

The PR also includes flags to configure all the possible context of the execution of the precompiles (i.e. whitelist, eth backend address, etc...).

Also, the configuration can be set from a configuration file in toml following the `profile` convention from Foundry files.

```
[profile.suave]
whitelist = ["*"]
eth_backend = "http:..."
```

## 📚 References

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

**Comment by lthibault** [comment]
Quick top-level question as I dive into this:  what is the intended use-case for this feature?  Local dev? Testing?

File: cmd/geth/forgecmd.go [hunk 1]
```diff
@@ -1,28 +1,129 @@
...
+
+	// create the suave context
+	var suaveEthBackend suave.ConfidentialEthBackend
+	if cfg.EthBackend != "" {
+		suaveEthBackend = suave_backends.NewRemoteEthBackend(cfg.EthBackend)
+	} else {
+		suaveEthBackend = &suave_backends.EthMock{}
+	}
+
+	ecdsaKey, _ := crypto.GenerateKey()
```
Comment (lthibault) [inline cmd/geth/forgecmd.go:99]: Can we check these errors?  It's fine to panic if we "know" they'll never fail.  I've just been bitten too many times by unchecked errors and have become a little neurotic as a result.

File: cmd/geth/forgecmd_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	for _, f := range flags {
+		if err := f.Apply(set); err != nil {
+			t.Fatal(err)
+		}
+	}
+	set.SetOutput(io.Discard)
+	return set
+}
+
+func TestForgeReadConfig(t *testing.T) {
```
Comment (lthibault) [inline cmd/geth/forgecmd_test.go:25]: Since this is a new test, I'd recommend `t.Parallel()`.

**Review by lthibault — CHANGES_REQUESTED** [review]

**Comment by ferranbt** [comment]
> Quick top-level question as I dive into this: what is the intended use-case for this feature? Local dev? Testing?

This is the feature that `suave-std` uses to execute the precompiles with forge. When Solidity tests in forge are executed, the VM that is used (anvil) does not have the precompiles from Suave. Thus, we add some Solidity "Connectors" in those addresses that make external calls to this binary to resolve the logic. This only works for stateless executions.

**Review by lthibault — APPROVED** [review]
LGTM!  Thanks for clarifying the purpose -- makes sense.

**Pull request merged**